AC_Fence: Remove argument to check_fence()

peterbarker opened the pull request:
Also rename it to just check().

Also remove unneeded initialisers.

OXINARF commented on libraries/AC_Fence/AC_Fence.cpp:
```diff
@@ -145,14 +145,20 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
 
+    Vector3f home;
+    if (_ahrs.get_relative_position_NED_home(home)) {
+        // we (may) remain breached if we can't update home
```
Humm, both the variable name and the comment suggest this method returns home, but it actually returns alt above home.
Also, any reason to not simply use `get_relative_position_D_home`? That's "guaranteed" to always return a value.

peterbarker commented:
On Fri, 15 Dec 2017, Francisco Ferreira wrote:

> @OXINARF commented on this pull request.
> 
> +    Vector3f home;
> +    if (_ahrs.get_relative_position_NED_home(home)) {
> +        // we (may) remain breached if we can't update home
> 
> Humm, both the variable name and the comment suggest this method returns
> home, but it actually returns alt above home.

It's all relative.... I think the Vector3f is the give-away for this 
myself.  Note that this is the same variable name as in the circle fence 
case... I think you can guess why.  Perhaps with the restructure mentioned 
below we can use a better name?  home -> vector_home?!  (doesn't say 
anything the type doesn't)....

> Also, any reason to not simply use get_relative_position_D_home? That's
> "guaranteed" to always return a value.

huh.  I'd forgotten about that one.  But I also thought briefly about 
saving a call onto AHRS and updating a private _home variable and 
_curr_alt *before* calling these check functions.  So having symmetry in 
our handling of getting an alt or not makes sense IMO.

I'm guessing we're going to have a baro-less vehicle at some stage.  I 
don't think we should continue to rely on having an altitude.  Of course, 
we're currently in the fence code.... but I think the _D_ call should die.

Yours,
-- 
Peter Barker                          |   Programmer,Sysadmin,Geek.
pbarker@barker.dropbear.id.au	      |   You need a bigger hammer.
:: It's a hack! Expect underscores! - Nigel Williams

OXINARF commented:
> It's all relative.... I think the Vector3f is the give-away for this
myself.  Note that this is the same variable name as in the circle fence
case... I think you can guess why.  Perhaps with the restructure mentioned
below we can use a better name?  home -> vector_home?!  (doesn't say
anything the type doesn't)....

I like something like `relpos_home`.

> huh.  I'd forgotten about that one.  But I also thought briefly about
saving a call onto AHRS and updating a private _home variable and
_curr_alt *before* calling these check functions.  So having symmetry in
our handling of getting an alt or not makes sense IMO.
>
> I'm guessing we're going to have a baro-less vehicle at some stage.  I
don't think we should continue to rely on having an altitude.  Of course,
we're currently in the fence code.... but I think the _D_ call should die.

I disagree with the symmetry, the methods need different things, there is no reason to request the same.
I have no issue with keeping the _curr_alt though as I hope someday the `_D_home` call will behave like the others and only return when it has a value instead of relying on baro as the fallback.

Why should the D call die? Should the NE methods die too? They are useful in several places, I don't see any reason not to have them.

peterbarker commented:
Renamed from `home` to `relpos_home`

OXINARF commented:
@peterbarker If you want to merge as is, you can either do it yourself or take it to a dev call. As I've wrote I disagree with using `get_relative_position_NED_home` where only `get_relative_position_D_home` is needed so I'm not merging it myself.

peterbarker commented:
On Wed, 27 Dec 2017, Francisco Ferreira wrote:

> get_relative_position_NED_home where only get_relative_position_D_home is
> needed so I'm not merging it myself.

I've moved this to call get_relative_position_D_home

OXINARF commented on libraries/AC_Fence/AC_Fence.h:
```diff
@@ -116,8 +115,11 @@ class AC_Fence
 
 private:
+
+    float curr_alt;
```
Humm, shouldn't this follow the other variables and have an underscore prefix? Putting it next to the `home_distance` also seems a good idea.

peterbarker commented:
On Wed, 27 Dec 2017, Francisco Ferreira wrote:

> Humm, shouldn't this follow the other variables and have an underscore
> prefix? Putting it next to the home_distance also seems a good idea.

Moved and renamed.

I preserved the name initially to reduce the size of the patches.

OXINARF commented:
Merged, thanks!
